### PR TITLE
Add support for all CrossSection (location) fields.

### LINKED ISSUE #172: Support all file fields for crosssection location files
Current implementation of crosssection locations (in `hydrolib.core.io.crosssection.models.CrossSection`) only has attributes `.id` and `.branchid`. We need to support the full file standard as listed in:
https://content.oss.deltares.nl/delft3d/manuals/D-Flow_FM_User_Manual_1D2D.pdf#subsection.C.16.2

Also include tests to check the optionally requiredness of some fields.

### PR DESCRIPTION
**Pull request by arthurvd**
Fixes #172.

### CODE REVIEW

File: hydrolib/core/io/crosssection/models.py [hunk 1]
```diff
@@ -639,14 +640,52 @@ class CrossSection(INIBasedModel):
 
     Attributes:
         id (str): Unique cross-section location id.
-        branchid (str): (optional) Branch on which the cross section is located.
+        branchid (str, optional): Branch on which the cross section is located.
+        chainage: (str, optional): Chainage on the branch (m).
+        x: (str, optional): x-coordinate of the location of the cross section.
+        y: (str, optional): y-coordinate of the location of the cross section.
```
Comment (priscavdsluis) [inline hydrolib/core/io/crosssection/models.py:646]:
```suggestion
        chainage (str, optional): Chainage on the branch (m).
        x (str, optional): x-coordinate of the location of the cross section.
        y (str, optional): y-coordinate of the location of the cross section.
```

File: hydrolib/core/io/crosssection/models.py [hunk 2]
```diff
@@ -639,14 +640,52 @@ class CrossSection(INIBasedModel):
...
-    branchid: str = Field(alias="branchId")
+
+    branchid: Optional[str] = Field(None, alias="branchId")
+    chainage: Optional[float] = Field(None)
+
+    x: Optional[float] = Field(None)
+    y: Optional[float] = Field(None)
+
+    branchid: Optional[str] = Field(alias="branchId")
+    shift: Optional[float] = Field(None)
```
Comment (priscavdsluis) [inline hydrolib/core/io/crosssection/models.py:683]: Default should be 0.0 as described in the manual

File: hydrolib/core/io/crosssection/models.py [hunk 3]
```diff
@@ -639,14 +640,52 @@ class CrossSection(INIBasedModel):
...
     id: str = Field(alias="id")
-    branchid: str = Field(alias="branchId")
+
+    branchid: Optional[str] = Field(None, alias="branchId")
+    chainage: Optional[float] = Field(None)
+
+    x: Optional[float] = Field(None)
+    y: Optional[float] = Field(None)
+
+    branchid: Optional[str] = Field(alias="branchId")
```
Comment (priscavdsluis) [inline hydrolib/core/io/crosssection/models.py:682]: This is the second branchid

File: hydrolib/core/io/ini/util.py [hunk 4]
```diff
@@ -313,7 +313,7 @@ def validate_coordinates(coord_name: str) -> None:
         if str_is_empty_or_none(node_id) and (str_is_empty_or_none(branch_id)):
             if n_coords == 0:
                 raise ValueError(
-                    f"Either {'nodeId, ' if allow_nodeid else ''}branchId (with chainage) or {numfield_name} (with {xfield_name} and {yfield_name}) are required."
```
Comment (priscavdsluis) [inline hydrolib/core/io/ini/util.py:316]: This has become a little hard to read, but we will update this anyway!

Comment (arthurvd) [reply hydrolib/core/io/ini/util.py:316]: I agree

File: hydrolib/core/io/crosssection/models.py (hunk 3 above)
Comment (arthurvd) [reply hydrolib/core/io/crosssection/models.py:682]: fixed

File: hydrolib/core/io/crosssection/models.py (hunk 2 above)
Comment (arthurvd) [reply hydrolib/core/io/crosssection/models.py:683]: I've fixed this. Indeed, consistent with all other *Models. Later we may want to reconsider: should defaults of the kernel be reflected in our Python classes? The alternative is to have None for most defaults, such that empty key-values are written in the produced input files, and leave it to the kernel. Food for thought later.

**Review by priscavdsluis — APPROVED** [review]

**Pull request merged**